packages/core/src/challenger/submitAssertionToReferee.ts

```diff
@@ -16,15 +18,39 @@ export async function submitAssertionToReferee(
     challengerBlsSecretKey: string,
     assertionId: number,
     assertionNode: AssertionNode,
-    signer: ethers.Signer
+    signer: ethers.Signer,
+    currentChallenge: Challenge
 ): Promise<void> {
 
+    // Get the assertion node for the current challenge
+    const lastSubmittedAssertionId = currentChallenge.assertionId;
+    const assertionIdGap = assertionId -  Number(lastSubmittedAssertionId);
+
+    const isBatch = assertionIdGap > 1;
+
+    let finalConfirmData: string;    
+    
+    if(isBatch) {
+
+        // Get the assertion IDs for the batch
+        const assertionIds = [...Array(assertionIdGap).keys()].map(i => i + Number(lastSubmittedAssertionId) + 1);
+
+        // Get the confirm data for all of the assertions
+        const [_, confirmDataHash] = await getMultipleChallengeConfirmData(assertionIds);
+
+        finalConfirmData = confirmDataHash;
+    }else{
+
+        finalConfirmData = assertionNode.confirmData;
+        
+    }
```

Comment:
This else can be removed if we assign `let finalConfirmData = assertionNode.confirmData` on declare.

Reply:
I appreciate you pointing this out, this has been updated. I will try to notice the opportunity to use this pattern in the future!